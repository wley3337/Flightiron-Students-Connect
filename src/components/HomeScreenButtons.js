import React from 'react'
import { Button } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import * as actions from '../redux/actions'

const HomeScreenButtons = (props) =>
    <div id="home-screen"> 
        
        <Button id="login" inverted as={ Link }  to="/login" onClick={props.deleteFlashMessage}>Log-In</Button>
        <Button id="create-account" inverted as={ Link }  to="/create-user" onClick={props.deleteFlashMessage}>Create Account</Button>

    </div>

export default connect(null, actions)(HomeScreenButtons)
