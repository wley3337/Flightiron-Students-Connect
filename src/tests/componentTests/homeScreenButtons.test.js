import React from 'react'
import { shallow } from 'enzyme'
//dependant components:
import HomeScreenButtons from '../../components/HomeScreenButtons'
import { Button, Item } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

describe('Home Screen Component Test: ', ()=>{

    const mockProps ={
        id: 'login',
        inverted: true,
        as: Link ,
        to: "/login",
        onClick: jest.fn()
    }
    const wrapper =  shallow(<Button {...mockProps}/>)

    it('renders consistantly', ()=>{
        expect(wrapper).toMatchSnapshot()
    })

    it('logs in onclick', ()=>{
        wrapper.find('#login').simulate('click');
        expect(mockProps.onClick).toHaveBeenCalled()
    })


})