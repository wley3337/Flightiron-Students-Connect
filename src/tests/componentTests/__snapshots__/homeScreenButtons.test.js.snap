// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Component Test:  renders consistantly 1`] = `
<Link
  className="ui inverted button"
  id="login"
  onClick={[Function]}
  replace={false}
  role="button"
  to="/login"
/>
`;
